neqo-transport/src/pmtud.rs

```diff
@@ -0,0 +1,180 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::{cell::RefCell, net::IpAddr, rc::Rc};
+
+use neqo_common::{qdebug, qtrace};
+
+use crate::{recovery::SentPacket, Stats};
+
+// From https://datatracker.ietf.org/doc/html/rfc1191#section-7.1, with a few modifications.
+const MTU_SIZES: [usize; 13] = [
+    65536, // Official maximum MTU          RFC 791
+    65535, // Hyperchannel                  RFC 1044
+    17914, // 16Mb IBM Token Ring           ref. [6]
+    16384, // macOS loopback
+    8166,  // IEEE 802.4                    RFC 1042
+    4464,  // IEEE 802.5 (4Mb max)          RFC 1042
+    4352,  // FDDI (Revised)                RFC 1188
+    2048,  // Wideband Network              RFC 907
+    2002,  // IEEE 802.5 (4Mb recommended)  RFC 1042
```

Comment:
The sizes from here on up are pretty uninteresting.  Do we really need 2002 and 2048?  The utility gain from an increase that small probably doesn't outweigh the cost of probing.